contracts/wallet/Proxy.sol

```diff
@@ -31,21 +32,20 @@ contract Proxy {
         implementation = _implementation;
     }
 
-    function() external payable {
-
-        if (msg.data.length == 0 && msg.value > 0) {
-            emit Received(msg.value, msg.sender, msg.data);
-        } else {
-            // solium-disable-next-line security/no-inline-assembly
-            assembly {
-                let target := sload(0)
-                calldatacopy(0, 0, calldatasize())
-                let result := delegatecall(gas, target, 0, calldatasize(), 0, 0)
-                returndatacopy(0, 0, returndatasize())
-                switch result
-                case 0 {revert(0, returndatasize())}
-                default {return (0, returndatasize())}
-            }
+    fallback() external payable {
+        // solium-disable-next-line security/no-inline-assembly
+        assembly {
+            let target := sload(0)
+            calldatacopy(0, 0, calldatasize())
+            let result := delegatecall(gas(), target, 0, calldatasize(), 0, 0)
+            returndatacopy(0, 0, returndatasize())
+            switch result
+            case 0 {revert(0, returndatasize())}
+            default {return (0, returndatasize())}
         }
     }
+
+    receive() external payable {
+        emit Received(msg.value, msg.sender, msg.data);
```

Comment:
Probably inconsequential but maybe worth notifying the clients that whereas before `Received` could not have been emitted with both `msg.value == 0` and `msg.data == 0`, it is now possible.

Reply:
I couldn't work out what's right for this event. [Here](https://github.com/argentlabs/argent-contracts/blob/fd1222b422cc2a1ad1c9e9518fe4f14717e9ffcf/contracts/wallet/BaseWallet.sol#L143) it is emitted for 0 values while in the Proxy it was not.

Reply:
Yes the previous implementation was confusing. I think it would have made more sense to check `if (msg.data.length == 0) { emit ... }` in the Proxy instead of `if (msg.data.length == 0 && msg.value > 0) { emit ... }`. This is equivalent to your new implementation, which I think is good.